services/datamanager/collection/collection.go

```diff
@@ -202,16 +210,23 @@ func GetCollections(params GetCollectionsParams) (results ReadCollections, err e
 		if err != nil {
 			return results, err
 		}
-		sdkParams["filters"].([][]string)[0][2] = latestModificationDate.Format(time.RFC3339)
+		if sdkParams["filters"].([][]string)[0][2] != latestModificationDate.Format(time.RFC3339) {
+			sdkParams["filters"].([][]string)[0][2] = latestModificationDate.Format(time.RFC3339)
+			sdkParams["offset"] = 0
+		} else {
+			sdkParams["offset"] = sdkParams["offset"].(int) + params.BatchSize
```

Comment:
I've made this change (using the aforementioned `batchCollections` rather than calling `len` multiple times), but for safety it also necessitates an additional sanity check that we did in fact receive some items in the batch. Otherwise, it would be theoretically possible to loop forever if the query returns no items (and no explicit error) but has non-zero ItemsAvailable. 

Maybe that is "not possible" given the current API server, but I think it is prudent to avoid a potential infinite loop and it doesn't cost much. 

Making the suggested change and adding the extra sanity check.